src/m3ninx/search/searcher/conjunction.go

```diff
@@ -72,13 +85,18 @@ func (s *conjunctionSearcher) Search(r index.Reader) (postings.List, error) {
 		}
 	}
 
+	lists = lists[:0]
 	for _, sr := range s.negations {
 		curr, err := sr.Search(r)
 		if err != nil {
 			return nil, err
 		}
+		lists = append(lists, curr)
+	}
 
-		// TODO: Sort the iterators so that we take the set differences in order of decreasing size.
+	sort.Sort(byLengthDescending(lists))
```

Comment:
I think specifically for difference, it makes sense to start from the smallest one, and subtract remaining ones in descending order.